actions/setup_jdk/action.yml

```diff
@@ -8,20 +8,63 @@ inputs:
   java-version:
     description: 'Java version'
     required: false
-    default: '19.0.2'
+    default: '20.0.2'
   distribution:
     description: 'Distribution (e.g. "adopt")'
     required: false
-    default: 'adopt' # Adopt OpenJDK
+    default: 'custom' # Adopt OpenJDK
+  downloadUrl:
+    description: 'URL to dwonload JDK archive. Should be used if `distribution` is custom.'
+    required: false
+    default: "https://download.java.net/java/GA/jdk20.0.2/6e380f22cbe7469fa75fb448bd903d8e/9/GPL/openjdk-20.0.2_linux-x64_bin.tar.gz"
+  checkSum:
+    description: 'SHA256 sum of archive.'
+    required: false
+    default: "beaf61959c2953310595e1162b0c626aef33d58628771033ff2936609661956c"
 
 runs:
   using: "composite"
   steps:
     - name: Set up JDK
       uses: actions/setup-java@v3.1.1
+      # this is needed for backward compatibility, download distributions 
+      # supported by actions/setup-java.
+      # `custom` is used to download actual openjdk
+      if: inputs.distribution != 'custom'
       with:
         java-version: ${{ inputs.java-version }}
         distribution: ${{ inputs.distribution }}
         # Do NOT overwrite Maven settings.xml with auto-generated one
         overwrite-settings: false
+      
+    - name: Download JDK
+      # `custom` is used to download by URL
+      if: inputs.distribution == 'custom'
+      shell: bash
+      run: |
+        download_url="${{ inputs.downloadUrl }}"
+        wget -q -O $RUNNER_TEMP/java_package.tar.gz $download_url     
```

Comment:
Ideally we would like to cache this download somewhere and not download it each time from official site, but it's pretty fast, so may work for now.

And working on caching it would require at least day for me, because I don't know how.

Reply:
About caching:

https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows
"When using self-hosted runners, caches from workflow runs are stored on GitHub-owned cloud storage. A customer-owned storage solution is only available with GitHub Enterprise Server."
So even if we use caching from GitHub actions, we would have to download file from their servers

Another possibility would be caching data in a local directory specific to our runner. Something like:
if (/avi-spl/cache/ dir exsists) {
  if file exists, use it
}
Two possibilities:
- safer option: have this file (and checksum) readonly: workflow just reads from it if present, otherwise downloads it in tmp dir. Files are added there before, once and for all by us/DevOps so we are sure the are valid.
- workflow writes file to this cache dir when downloaded

I think we can live with it for now, and create a separate task for that